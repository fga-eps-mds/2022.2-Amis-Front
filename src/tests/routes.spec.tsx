import { QueryClient, QueryClientProvider } from "react-query";
import { ThemeProvider } from "styled-components";
import AppRoutes from "../routes";
import theme from "../styles/theme";
import { render } from "@testing-library/react";




const renderComponent = async()=> {
  const queryClient = new QueryClient ();
  render(
    // eslint-disable-next-line react/react-in-jsx-scope
    <QueryClientProvider client={queryClient}>
          <ThemeProvider theme={theme}>
            <AppRoutes />
          </ThemeProvider>
    </QueryClientProvider>
  );
  return queryClient;
}

describe("AppRoutes", () => {
  it("Teste para renderizar o componente",  () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    renderComponent();
  });
});
