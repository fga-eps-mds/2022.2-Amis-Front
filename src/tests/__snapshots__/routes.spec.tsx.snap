// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Deve corresponder ao Snapshot 1`] = `
<div>
  <div
    className="sc-ipEyDJ iNhSUS"
  >
    <div
      className="sc-csuSiG hOMzVq"
    >
      <div
        className="sc-eDWCr fPDczd"
      >
        <h1
          className="sc-bqWxrE bsUfAj"
        >
          AMIS
        </h1>
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="sc-ksBlkl cHvHkv"
            onClick={[Function]}
          >
            Home
          </button>
        </a>
        <a
          href="/receitas"
          onClick={[Function]}
        >
          <button
            className="sc-ksBlkl cgyygN"
            onClick={[Function]}
          >
            Receitas
          </button>
        </a>
      </div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Área Logada
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-bjfHbI blaGIl"
  >
    <div
      className="sc-idXgbr jCQzeu"
    >
      <h1
        className="sc-hHTYSt jDkCJe"
      >
        AMIS
      </h1>
      <h1
        className="sc-hHTYSt jDkCJe"
      >
        Mulheres Criativas
      </h1>
      <span
        style={
          {
            "paddingTop": "30px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
      <span
        style={
          {
            "paddingBottom": "30px",
          }
        }
      >
        Varius sed pharetra dictum neque massa congue
      </span>
      <div
        style={
          {
            "display": "flex",
            "gap": "15px",
          }
        }
      >
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Saiba Mais
        </button>
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Faça um pedido
        </button>
      </div>
    </div>
    <div
      style={
        {
          "position": "relative",
        }
      }
    >
      <img
        className="sc-dIfARi dvNEGs"
        src="test-file-stub"
        style={
          {
            "bottom": -80,
            "left": -200,
            "position": "absolute",
            "width": "300px",
            "zIndex": 5,
          }
        }
      />
      <img
        className="sc-dIfARi dvNEGs"
        src="test-file-stub"
        style={
          {
            "width": "400px",
            "zIndex": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-dmctIk eqmaEq"
  >
    <h1
      className="sc-kgTSHT ihVGqN"
    >
      Quem Somos
    </h1>
    <span
      className="sc-fLcnxK eMddWG"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam aliquam egestas ultricies. Morbi venenatis, odio a laoreet vulputate, sapien tellus ornare orci, sit amet feugiat arcu dui sed lorem. Maecenas et lacus vel lacus lacinia porttitor. Nulla tempus dictum efficitur. Suspendisse potenti. Sed ac nulla eget nunc tempor faucibus sed et mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse potenti. Quisque tortor ipsum, tincidunt eget ligula at, aliquet commodo purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; In pretium leo ac dolor gravida pretium. Sed eu venenatis lacus, efficitur pulvinar massa. Morbi aliquet sapien vitae purus tincidunt, eget consequat erat iaculis. Donec ut sem ac sapien laoreet congue id et neque. Ut fermentum non libero ut consequat. Quisque elementum velit eget ipsum suscipit rutrum.
    </span>
    <h1
      className="sc-kgTSHT ihVGqN"
    >
       Nossa missão
    </h1>
    <span
      className="sc-fLcnxK eMddWG"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam aliquam egestas ultricies. Morbi venenatis, odio a laoreet vulputate, sapien tellus ornare orci, sit amet feugiat arcu dui sed lorem. Maecenas et lacus vel lacus lacinia porttitor. Nulla tempus dictum efficitur. Suspendisse potenti. Sed ac nulla eget nunc tempor faucibus sed et mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse potenti. Quisque tortor ipsum, tincidunt eget ligula at, aliquet commodo purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; In pretium leo ac dolor gravida pretium. Sed eu venenatis lacus, efficitur pulvinar massa. Morbi aliquet sapien vitae purus tincidunt, eget consequat erat iaculis. Donec ut sem ac sapien laoreet congue id et neque. Ut fermentum non libero ut consequat. Quisque elementum velit eget ipsum suscipit rutrum.
    </span>
  </div>
  <div
    className="sc-bBABsx hdTJTu"
  >
    <div
      className="sc-iveFHk kxAKQE"
    >
      <h1
        className="sc-cwSeag dKRnpi"
      />
      <span
        className="sc-lllmON bUzVLa"
      >
        Mulheres atendidas
      </span>
    </div>
    <div
      className="sc-iveFHk kxAKQE"
    >
      <h1
        className="sc-cwSeag dKRnpi"
      />
      <span
        className="sc-lllmON bUzVLa"
      >
        Formações profissionais
      </span>
    </div>
  </div>
  <div
    className="sc-iJnaPW jOPtMN"
  >
    <div>
      <h1
        className="sc-gikAfH dNLHga"
      >
        Nossas receitas
      </h1>
    </div>
  </div>
  <div
    className="sc-hBxehG ihUWmk"
  >
    <a
      href="https://www.instagram.com/amismulherescriativas/"
    >
      <img
        className="sc-fnGiBr inBseq"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20x",
            "width": "40px",
          }
        }
      />
    </a>
    <a
      href="https://pt-br.facebook.com/InstitutoMulheresCriativas/"
    >
      <img
        className="sc-fnGiBr inBseq"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20px",
            "width": "20px",
          }
        }
      />
    </a>
    <div>
      <span
        className="sc-fEXmlR cWpkJD"
      >
        © 2022. All rights reserved.
      </span>
    </div>
  </div>
</div>
`;
